// etsy-js/lib/etsyjs/shipping.js
(function() {
  var Shipping,
    __slice = [].slice;

  Shipping = (function() {
    function Shipping(shippingId, client) {
      this.shippingId = shippingId;
      this.client = client;
    }

    Shipping.prototype.find = function(cb) {
      return this.client.get("/shipping/templates/" + this.shippingId, function(err, status, body, headers) {
        if (err) {
          return cb(err);
        }
        if (status !== 200) {
          return cb(new Error('Get shipping templates error'));
        } else {
          return cb(null, body, headers);
        }
      });
    };

    Shipping.prototype.createshipping = function(shipping,cb) {
      //console.log(this);
      console.log(shipping);
      return this.client.post("/shippings", shipping,function(err, status, body, headers) {
        if (err) {
          return cb(err);
        }
        if (status !== 200) {
          return cb(new Error('Create shippings error'));
        } else {
          return cb(null, body, headers);
        }
      });
    };

    Shipping.prototype.findAllShippingTemplates = function(cb) {
      return this.client.get("/users/"+ this.userId +"/shipping/templates", function(err, status, body, headers) {
        if (err) {
          return cb(err);
        }
        if(status !== 200) {
          return cb(new Error("Get shipping templates error"));
        } else {
          return cb(null, body, headers);
        }
      })
    }

    return Shipping;

  })();

  module.exports = Shipping;

}).call(this);
